lightning/src/chain/channelmonitor.rs

```diff
@@ -2116,13 +2345,22 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 						source: source.clone(),
 						onchain_value_satoshis,
 					}));
+					if let Some(idx) = input_idx {
+						self.htlcs_resolved_on_chain.push(HTLCIrrevocablyResolved { input_idx: idx, payment_preimage: None });
+					}
 				},
 				OnchainEvent::MaturingOutput { descriptor } => {
 					log_debug!(logger, "Descriptor {} has got enough confirmations to be passed upstream", log_spendable!(descriptor));
 					self.pending_events.push(Event::SpendableOutputs {
 						outputs: vec![descriptor]
 					});
-				}
+				},
+				OnchainEvent::HTLCSpendConfirmation { input_idx, our_preimage, .. } => {
+					self.htlcs_resolved_on_chain.push(HTLCIrrevocablyResolved { input_idx, payment_preimage: our_preimage });
+				},
+				OnchainEvent::FundingSpendConfirmation { txid, .. } => {
+					self.funding_spend_confirmed = Some(txid);
```

Comment:
Hm, I don't think we currently reset this to `None` if the funding tx is reorged?

Reply:
True, though we should only get to this point after `ANTI_REORG_DELAY`, and at that point we provide no guarantees of anything working correctly.